pmlc/conversion/stdx_to_llvm/stdx_to_llvm.cc

```diff
@@ -98,6 +100,79 @@ struct FPToUILowering : public LLVMLegalizationPattern<stdx::FPToUIOp> {
   }
 };
 
+class BaseViewConversionHelper {
+public:
+  explicit BaseViewConversionHelper(Type type)
+      : desc(MemRefDescriptor::undef(rewriter(), loc(), type)) {}
+
+  explicit BaseViewConversionHelper(Value v) : desc(v) {}
+
+  /// Wrappers around MemRefDescriptor that use EDSC builder and location.
+  Value allocatedPtr() { return desc.allocatedPtr(rewriter(), loc()); }
+  void setAllocatedPtr(Value v) { desc.setAllocatedPtr(rewriter(), loc(), v); }
+  Value alignedPtr() { return desc.alignedPtr(rewriter(), loc()); }
+  void setAlignedPtr(Value v) { desc.setAlignedPtr(rewriter(), loc(), v); }
+  Value offset() { return desc.offset(rewriter(), loc()); }
+  void setOffset(Value v) { desc.setOffset(rewriter(), loc(), v); }
+  Value size(unsigned i) { return desc.size(rewriter(), loc(), i); }
+  void setSize(unsigned i, Value v) { desc.setSize(rewriter(), loc(), i, v); }
+  void setConstantSize(unsigned i, int64_t v) {
+    desc.setConstantSize(rewriter(), loc(), i, v);
+  }
+  Value stride(unsigned i) { return desc.stride(rewriter(), loc(), i); }
+  void setStride(unsigned i, Value v) {
+    desc.setStride(rewriter(), loc(), i, v);
+  }
+  void setConstantStride(unsigned i, int64_t v) {
+    desc.setConstantStride(rewriter(), loc(), i, v);
+  }
+
+  operator Value() { return desc; }
+
+private:
+  OpBuilder &rewriter() { return edsc::ScopedContext::getBuilder(); }
+  Location loc() { return edsc::ScopedContext::getLocation(); }
+
+  MemRefDescriptor desc;
+};
+
+struct ReshapeLowering : public LLVMLegalizationPattern<stdx::ReshapeOp> {
+  using LLVMLegalizationPattern<stdx::ReshapeOp>::LLVMLegalizationPattern;
+  using Base = LLVMLegalizationPattern<stdx::ReshapeOp>;
+
+  LogicalResult
+  matchAndRewrite(Operation *op, ArrayRef<Value> operands,
+                  ConversionPatternRewriter &rewriter) const override {
+    auto reshapeOp = cast<stdx::ReshapeOp>(op);
+    MemRefType dstType = reshapeOp.getResult().getType().cast<MemRefType>();
```

Comment:
I'd just add here a defensive check to make sure reshapeOp is not nullptr. Other than that it looks fine to me.

Reply:
We're already ensuring the type is `stdx::ReshapeOp` because the base class is templated on this type. The `cast` already has an assertion inside of it, so no need for additional checks.